client/src/app/app.component.ts

```diff
@@ -26,7 +28,12 @@ export class AppComponent implements OnInit{
                     `http://${environment.API_HOST}:${environment.API_PORT}/sessions/${this.ID}`
                 )
                 .subscribe((response) => {
-                    if (response) this.loggedIn = true
+                    if (response) { 
```

Comment:
Instead of making it only do it if the response is true, maybe check if the current state is not logged  in and it changed, and only if it changed run one of these.